mono/utils/mono-threads-api.h

```diff
@@ -130,13 +130,11 @@ For further explanation of what can and can't be done in GC unsafe mode:
 	} while (0)
 
 #define MONO_ENTER_GC_SAFE_UNBALANCED	\
-	do {	\
 		MONO_STACKDATA (__gc_safe_unbalanced_dummy); \
 		gpointer __gc_safe_unbalanced_cookie = mono_threads_enter_gc_safe_region_unbalanced_internal (&__gc_safe_unbalanced_dummy)
 
 #define MONO_EXIT_GC_SAFE_UNBALANCED	\
-		mono_threads_exit_gc_safe_region_unbalanced_internal (__gc_safe_unbalanced_cookie, &__gc_safe_unbalanced_dummy);	\
-	} while (0)
+		mono_threads_exit_gc_safe_region_unbalanced_internal (__gc_safe_unbalanced_cookie, &__gc_safe_unbalanced_dummy);
```

Comment:
it seems like this macro has almost no value: it won't compile unless you pair it with `MONO_ENTER_GC_SAFE_UNBALANCED` (because it needs the cookie).  But if you use it like that, it's balanced!